content/projects/gsoc/2020/projects/custom-jenkins-distribution-build-service.adoc

```diff
@@ -20,7 +20,52 @@ tags:
+The main idea behind the project is to build a customizable Jenkins distribution service that could be used to build tailor-made Jenkins distributions. The service would provide users with a simple interface to select the configurations they want to build the instance with eg: plugins, authorization matrices etc. Furthermore it would include a section for sharing community created distros so that users can find and download already built Jenkins war/configuration files to use out of the box.
+
+=== Key Deliverables
+* A backend service built on spring boot having a modern Restful API support for constructing a  customized WAR package and configuration files.
+* A configuration constructing engine that can convert UI fields into a configuration file.
+* Front End application in the form of a website with a simple user interface for the users to interact with and customize the distribution.
+* Repository sharing community built most commonly used Jenkins configuration files.
+
+=== Plan for Phase 1 
+
+* Develop a simple workflow(Spring Boot Application) from UI bindings to the generation of the JCasC YAML file.
+  ** Reuse concepts from the JCasC Plugin to generate this JCasC file. 
+  ** Verify the configuration generated against the JCasC Schema.
+* Begin with a simple User Interface to test the workflow against.
+* Reuse logic from the jenkins core to construct and if possible render the user interface in react for all the plugins.
+* If the rendering is not possible, default to building the schemas from the data bound constructors or the JCasC Schema.
+* Blog Post for Phase 1.
+
+=== Plan for Phase 2
+
+* Utilize the custom war packager library for heavy lifting and converting the configuration file to a war package.
+* This project builds a wrapper around the CWP so some logic can be reused from here if necessary.
+* Integration of the above  wrapper into the service.
+ ** This step should be relatively simple since the CWP library is fairly straightforward.
+* The service should really take shape after this week since we would have build the entire workflow from:
+UI -> Config -> War Package
+* Work on hosting the service.
+* Blog Post for Phase 2.
+
+=== Plan for Phase 3
+
+* Work on community hosted repositories for most used configuration files.
+* Set up a github repository(Source of Truth) with the configs, allowing users to share their configs either via the service
+* Workflow supported: Service -> Github
+* Login feature for users  via Github or jenkins.io.
+* Work on developing the database schema to store the user generated configurations.
+* Integration of database into the service.
```

Comment:
Subject for a discussion at the project meeting. I do not think we should be storing private configurations on our side, would be better to offer a self-hosted option for it

Reply:
Considering users can deploy this project to their own private environment, it's necessary. But we can reduce the priority of this feature.

Reply:
I included it in the proposal as an additional feature,we can discuss more about it in the project meeting.